omdb/index.js

```diff
@@ -0,0 +1,108 @@
+var express = require('express');
+var ejsLayouts = require('express-ejs-layouts');
+var request = require('request');
+var app = express();
+var db = require('./models'); //every model represents a table in our table base
+var bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({extended:false}));
+
+app.set('view engine', 'ejs');
+app.use(ejsLayouts);
+app.use(express.static(__dirname + '/static'));
+
+
+
+
+app.get('/', function(req, res) {
+  res.render('index');
+});
+
+app.get('/movies/favorites', function(req, res) {// does this have to be favorites? btw "x/x/x" is called route
+  db.favorite.findAll().then(function(favs) {
+    res.render('favPage', { // this is the visial page that gets the info dumped into
+      allFavs: favs //this is where the app.post info goes 
+    });
+  });
+});
+
+app.get('/movies', function(req, res) { 
+  var query = req.query.q;
+  request('http://www.omdbapi.com/?s=' + query, function(err, response, body) {
+    var data = JSON.parse(body);
+    if (!err && response.statusCode === 200 && data.Search) {
+      res.render('movies', {movies: data.Search,
+                            q: query});
+    } else {
+      res.render('error');
+    }
+  });
+});
+
+app.get('/movies/:imdbID', function(req, res) {
+  var searchQuery = req.query.q ? req.query.q : '';
+  var imdbID = req.params.imdbID;
+  request('http://www.omdbapi.com/?i=' + imdbID, function(err, response, body) {
+    res.render('movieShow', {movie: JSON.parse(body),
+                             q: searchQuery});
+  });
+});
+
+app.post('/getFavs', function(req, res) {
```

Comment:
Hi Cassie, for this route, we need to use findOrCreate. The difference is that if we just use create, it will look for the movie you want to add, check if it's there, and if it's there it will make a duplicate of that record (this is bad, think about having 50 records for the same movie, which is a huge waste of memory!). FindOrCreate will look for the movie you want to add, check if it's there, and if it's not, it will create the movie in the database, if it's there, it will return the movie to you.
```
app.post('/getFavs', function(req, res) {
  db.favorite.findOrCreate({
   where: {
      imdb_code: req.body.imbd_id,
      title: req.body.title,
      year: req.body.year
    }
  }).then(function(created) {
    res.redirect('/movies/favorites');
  });
});